packages/core/src/test/notifications/rendering.test.ts

```diff
@@ -0,0 +1,215 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as sinon from 'sinon'
+import assert from 'assert'
+import { ToolkitNotification } from '../../notifications/types'
+import { panelNode } from './controller.test'
+import { getTestWindow } from '../shared/vscode/window'
+import * as VsCodeUtils from '../../shared/utilities/vsCodeUtils'
+import { readonlyDocument } from '../../shared/utilities/textDocumentUtilities'
+
+describe('Notifications Rendering', function () {
+    let sandbox: sinon.SinonSandbox
+    //const panelNode: NotificationsNode = testNotificationsNode
+
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    // util to test txt pop-up under different senarios
+    async function verifyTxtNotification(notification: ToolkitNotification) {
+        const expectedContent = notification.uiRenderInstructions.content['en-US'].description
+        const readonlyDocumentShowStub = sandbox.stub(readonlyDocument, 'show').resolves()
+
+        await panelNode.openNotification(notification)
+
+        assert.ok(readonlyDocumentShowStub.calledOnce)
+        assert.ok(readonlyDocumentShowStub.calledWith(expectedContent, `Notification: ${notification.id}`))
+    }
+
+    // util to test open url under different senarios
+    async function verifyOpenExternalUrl(notification: ToolkitNotification) {
+        const url = vscode.Uri.parse('https://aws.amazon.com/visualstudiocode/')
+        //const openExternalStub = getOpenExternalStub()
+        const openUrlStub = sandbox.stub(VsCodeUtils, 'openUrl')
+        await panelNode.openNotification(notification)
+
+        assert.ok(openUrlStub.calledOnce)
+        assert.ok(openUrlStub.calledWith(url))
+    }
```

Comment:
This is a lot of code just to check that a url was clicked. 

One benefit of having shared functions such as `openUrl`, is that we don't need to write tests for them over and over. If a button only calls openUrl, we don't need tests for that.

If there is other state that we can check (and the function can return as a result) when the button is clicked, test for that instead.

Reply:
`getTestWindow` may already have stuff that can check that the selection was clicked. Other than that, we don't need to check that `openUrl` opens a URL.